packages/block-library/src/navigation/index.php

```diff
@@ -390,49 +390,20 @@ function block_core_navigation_get_most_recently_published_navigation() {
 }
 
 /**
- * Recursively filter out blocks from the block list that are not whitelisted.
- * This list of exclusions includes:
- * - The Navigation block itself (results in recursion).
- * - empty "null" blocks from the block list.
- * - other blocks that are not yet handled.
- *
- * Note: 'parse_blocks' includes a null block with '\n\n' as the content when
+ * Filter out empty "null" blocks from the block list.
+ * 'parse_blocks' includes a null block with '\n\n' as the content when
  * it encounters whitespace. This is not a bug but rather how the parser
  * is designed.
  *
- * @param array $parsed_blocks the parsed blocks to be filtered.
- * @return array the filtered parsed blocks.
+ * @param array $parsed_blocks the parsed blocks to be normalized.
+ * @return array the normalized parsed blocks.
  */
-function block_core_navigation_filter_out_invalid_blocks( $parsed_blocks ) {
-	// This list is duplicated in /packages/block-library/src/navigation/edit/inner-blocks.js.
-	$allowed_blocks = array(
-		'core/navigation-link',
-		'core/search',
-		'core/social-links',
-		'core/social-link',
-		'core/page-list',
-		'core/spacer',
-		'core/home-link',
-		'core/site-title',
-		'core/site-logo',
-		'core/navigation-submenu',
-	);
-
-	$filtered = array_reduce(
+function block_core_navigation_filter_out_empty_blocks( $parsed_blocks ) {
```

Comment:
Should we change this function name?

Reply:
The name was reverted to what it was called prior to https://github.com/WordPress/gutenberg/pull/46279